purchases/src/main/kotlin/com/revenuecat/purchases/PostReceiptHelper.kt

```diff
@@ -102,13 +103,23 @@ internal class PostReceiptHelper(
             storeUserID = purchase.storeUserID,
             marketplace = purchase.marketplace,
             initiationSource = initiationSource,
-            onSuccess = { info ->
-                billing.consumeAndSave(finishTransactions, purchase, initiationSource)
-                onSuccess?.let { it(purchase, info) }
+            onSuccess = { postReceiptResponse ->
+                // Currently we only support a single token per postReceipt call but multiple product Ids.
+                // The backend would fail if given more than one product id (multiline purchases which are
+                // not supported) so it's safe to pickup the first one.
+                // We would need to refactor this if/when we support multiple tokens per call.
```

Comment:
I guess one important thing to keep in mind here is that if we did add backend-side support, this SDK would remain unchanged in the wild. We can update it, but many devs will continue using this version which mistakenly refers to our backend not working

Reply:
It does make me wonder, given that consumption really happens at the token level, whether we should have had the consumption field be top-level in post-receipt response so that it unequivocally tells you whether or not to consume

Reply:
I mean, this SDK won't work with multiline purchases. As you said, currently, we only support sending one token per request. The idea of this response was to future-proof it so, in case we add support for multiple tokens/products per request, we don't need to change the response for this data.

> the consumption field be top-level in post-receipt response

We can do that, but as I mentioned, if we add support for multiple tokens per request, we would need to change this part of the response. We might be doing this too early, but I liked giving it more flexibility like this, however, it's completely unneeded right now yeah. Also, if we did that, I would still add it into a `purchased_product` field or similar so it's clear that it's information about the product, but only have one product in the response, in case we want to add other info about the product in the future.

Lmk if you feel strongly about it and I can still change the backend side.